kv/kvserver: TestReadLoadMetricAccounting failed

kv/kvserver.TestReadLoadMetricAccounting [failed](https://tanzanite.cluster.engflow.com/invocations/default/d5e88dd5-4855-4f82-9a2a-75631bf6d86c?testReportRun=3&testReportShard=26&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on release-25.1 @ [0a31e426c253fe576966dd943b3d9b4897b1137c](https://github.com/cockroachdb/cockroach/commits/0a31e426c253fe576966dd943b3d9b4897b1137c):


```
=== RUN   TestReadLoadMetricAccounting
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestReadLoadMetricAccounting3193842188
    test_log_scope.go:76: use -show-logs to present logs inline
    replica_rankings_test.go:206: 
        	Error Trace:	pkg/kv/kvserver/replica_rankings_test.go:206
        	            				pkg/kv/kvserver/replica_rankings_test.go:500
        	Error:      	Max difference between 0 and 89 allowed is 4, but difference was -89
        	Test:       	TestReadLoadMetricAccounting
    panic.go:626: -- test log scope end --
test logs left over in: outputs.zip/logTestReadLoadMetricAccounting3193842188
--- FAIL: TestReadLoadMetricAccounting (4.28s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=3</code>
 - <code>shard=26</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReadLoadMetricAccounting.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47916

--- a/pkg/kv/kvserver/replica_rankings_test.go
+++ b/pkg/kv/kvserver/replica_rankings_test.go
@@ -14,10 +14,12 @@ import (
 	"testing"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/keys"
 	"github.com/cockroachdb/cockroach/pkg/kv"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvpb"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/allocator"
 	aload "github.com/cockroachdb/cockroach/pkg/kv/kvserver/allocator/load"
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/kvserverbase"
 	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/load"
 	"github.com/cockroachdb/cockroach/pkg/roachpb"
 	"github.com/cockroachdb/cockroach/pkg/storage"
@@ -26,6 +28,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/testutils/storageutils"
 	"github.com/cockroachdb/cockroach/pkg/util/leaktest"
 	"github.com/cockroachdb/cockroach/pkg/util/log"
+	"github.com/cockroachdb/logtags"
 	"github.com/stretchr/testify/require"
 )
 
@@ -390,6 +393,32 @@ func TestReadLoadMetricAccounting(t *testing.T) {
 
 	tc := serverutils.StartCluster(t, 1, base.TestClusterArgs{
 		ReplicationMode: base.ReplicationManual,
+		ServerArgs: base.TestServerArgs{Knobs: base.TestingKnobs{
+			Store: &StoreTestingKnobs{
+				EvalKnobs: kvserverbase.BatchEvalTestingKnobs{
+					TestingPostEvalFilter: func(args kvserverbase.FilterArgs) *kvpb.Error {
+						if !args.Req.Header().Span().Overlaps(roachpb.Span{
+							Key: keys.ScratchRangeMin, EndKey: keys.ScratchRangeMax,
+						}) {
+							return nil
+						}
+
+						buf := logtags.FromContext(args.Ctx)
+						if buf != nil {
+							buf = &logtags.Buffer{}
+						}
+						if reflect.TypeOf(args.Req) == reflect.TypeOf(&kvpb.AddSSTableRequest{}) {
+							t.Logf("evaluated [logtags=%s: %T", buf, args.Req)
+						} else {
+							// Something unknown we likely did not expect.
+							t.Logf("evaluated [logtags=%s]: %T on %s: %s %+v",
+								buf, args.Req, args.Req.Header().Span(), args.Req, args.Hdr)
+						}
+						return nil
+					},
+				},
+			},
+		}},
 	})
 
 	defer tc.Stopper().Stop(ctx)
@@ -469,7 +498,8 @@ func TestReadLoadMetricAccounting(t *testing.T) {
 	// artificially inflating measurement due to consistency checking.
 	sqlDB.Exec(t, `SET CLUSTER SETTING server.consistency_check.interval = '0'`)
 
-	for _, testCase := range testCases {
+	for i, testCase := range testCases {
+		t.Logf("test #%d", i+1)
 		// Reset the request counts to 0 before sending to clear previous requests.
 		repl.loadStats.Reset()
 

